### PR TITLE
Add maestro-outlet and maestro-modal-once components plus Button styles

### PR DESCRIPTION
**Pull request by BradDenver**
Trello: https://trello.com/c/v5a9flnn/524-modal-on-sitepoint-forums

follows on from https://github.com/sitepoint/blog/pull/851 and https://github.com/sitepoint/sp-maestro/pull/51

Ember has a concept of outlets where components can be inserted via connectors eg. `assets/javascripts/discourse/templates/connectors/below-footer/maestro.hbs` will cause a `maestro-outlet` component to be inserted into the `plugin-below-footer` outlet.

`{{maestro-outlet templateAlias="Forums Below Footer"}}`
The `maestro-outlet` component will fetch a template from maestro for the given `templateAlias`/`widget name` and then render it. As it is rendered by Ember rather than just inserting into the dom the template can contain other Ember components (eg maestro-modal-once).

The `maestro-modal-once` component creates a modal window. The window is not shown until `activateDelay` seconds have elapsed (the component could be extended to activate on scroll in the future. The component also accepts a `trackingLabel` property that is used for both Google Analytics and local storage. As with the SlideUp component on blog local storage is used to store when a user has seen the modal and the modal is not shown again for 30 days. The modal window uses Discourses standard modal component but adds a `.maestro-modal-once` css class to allow custom styling.

screen shot 2016-06-09 at 12 33 10 pm

Example template from maestro
```
<script type="text/x-handlebars">
{{#maestro-modal-once activateDelay=3 trackingLabel="test"}}
    <form method="get" action="https://www.sitepoint.com/premium/free_ebooks" style="padding: 1em; text-align: center;">
        <input type="hidden" name="utm_source" value="sitepoint" />
        <input type="hidden" name="utm_medium" value="slideup" />
        <input type="hidden" name="utm_campaign" value="ebookbundle" />
        <div style="font-size: 48px; font-weight: bold; line-height: 1.3;">10 free SitePoint ebooks</div>
        <div class="emoji" style="font-size: 24px; font-weight: 300; line-height: 1.3; color: #6f6f6f;"> Just because we like you </div>
        <div style="margin-top: 4em; margin-bottom: 1em;">
          <input class="Button u-large u-cta" type="submit" value="Claim Your Ebooks Now!" />
        </div>
        <div style="font-weight: 300; line-height: 1.3; color: #a2a2a2;">Totally free. No credit card required.</div>
        <img style="max-width: 100%; margin-top: 4em;" src="https://dab1nmslvvntp.cloudfront.net/wp-content/uploads/2016/06/1465433902free-books.png" />
    </form>
    <style>
      .maestro-modal-once { font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; }
      .maestro-modal-once .modal-header { border: none; }
      .maestro-modal-once .modal-inner-container { max-width: 900px; }
      .maestro-modal-once .emoji::before { content: ' \1f4da'; }
      .maestro-modal-once .emoji::after { content: '\1f604'; }
    </style>
{{/maestro-modal-once}}
</script>
```

### CODE REVIEW

**Comment by mrlagmer** [comment]
👍

File: assets/javascripts/discourse/components/maestro-modal-once.js.es6 [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  click (e) {
+    const $target = $(e.target);
+    if ($target.hasClass("modal-middle-container") ||
+        $target.hasClass("modal-outer-container")) {
+      this.track("Background");
+      this.set("shouldRender", false);
+    }
+  },
+
+	hasUserSeen () {
```
Comment (santouras) [inline assets/javascripts/discourse/components/maestro-modal-once.js.es6]: indenting looks weird here

File: assets/javascripts/discourse/templates/components/maestro-modal-once.hbs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+<div class="modal maestro-modal-once">
+  <div class="modal-outer-container">
+    <div class="modal-middle-container">
+      <div class="modal-inner-container">
+        <div class="modal-header">
+          <a class="close" {{action "closeModal"}}>{{fa-icon "times"}}</a>
+          <h3>{{title}}</h3>
+          <div class="clearfix"></div>
+        </div>
+				<div class="modal-body full-height-modal">{{yield}}</div>
```
Comment (santouras) [inline assets/javascripts/discourse/templates/components/maestro-modal-once.hbs]: indenting looks weird here

File: assets/javascripts/discourse/components/maestro-outlet.js.es6 [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  hasRendered: false,
+
+  _initMaestroOutlet: function() {
+    const comp = this;
+    if (this.hasRendered) return;
+    if (!this.templateAlias) return console.log("No templateAlias supplied for maestro outlet", this);
+
+    $.ajax({
+      type: 'POST',
+      url: `/wp-admin/admin-ajax.php`,
```
Comment (santouras) [inline assets/javascripts/discourse/components/maestro-outlet.js.es6]: does this need to be backticks?

**Comment by santouras** [comment]
👌

**Pull request merged**